### PR TITLE
Use a new Skia API for safe passing of images across threads and GrContexts

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/9439

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

**Comment by Hixie** [comment]
Is this the kind of thing we can test?

File: lib/ui/painting/image_decoding.cc [hunk 1]
```diff
@@ -34,24 +34,13 @@ sk_sp<SkImage> DecodeImage(sk_sp<SkData> buffer) {
...
-
-  if (raster_image == nullptr) {
-    return nullptr;
-  }
-
-  if (auto context = ResourceContext::Get()) {
-    // TODO: Supply actual destination color space once available
-    if (auto texture_image =
-            raster_image->makeTextureImage(context, nullptr)) {
-#ifdef OS_ANDROID
```
Comment (chinmaygarde) [inline lib/ui/painting/image_decoding.cc]: There is an import for `GLES2/gl2.h` on top that you should probably get rid of too.

File: lib/ui/painting/image_decoding.cc [hunk 2]
```diff
@@ -34,24 +34,13 @@ sk_sp<SkImage> DecodeImage(sk_sp<SkData> buffer) {
...
   }
 
-  auto raster_image = SkImage::MakeFromEncoded(std::move(buffer));
-
-  if (raster_image == nullptr) {
-    return nullptr;
-  }
-
-  if (auto context = ResourceContext::Get()) {
-    // TODO: Supply actual destination color space once available
```
Comment (chinmaygarde) [inline lib/ui/painting/image_decoding.cc]: Can be keep this comment please? Adding it next to the last argument of `SkImage::MakeCrossContextFromEncoded`.

Comment (jason-simmons) [inline lib/ui/painting/image_decoding.cc]: done

File: lib/ui/painting/image_decoding.cc (hunk 1 above)
Comment (jason-simmons) [inline lib/ui/painting/image_decoding.cc]: done

**Comment by chinmaygarde** [comment]
> Is this the kind of thing we can test?

This will fix drawing of atlases (and not crashing on an assert in debug engine builds). We need pixel tests for that.

**Pull request merged**